dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractJdbcEventAnalyticsManager.java

```diff
@@ -134,19 +141,29 @@ private String getSortClause( EventQueryParams params )
 
         if ( params.isSorting() )
         {
-            sql += "order by ";
+            sql += "order by " + getSortColumns( params, "asc" ) + getSortColumns( params, "desc" );
 
-            for ( DimensionalItemObject item : params.getAsc() )
+            sql = TextUtils.removeLastComma( sql ) + " ";
+        }
+
+        return sql;
+    }
+
+    private String getSortColumns(EventQueryParams params , String order) {
```

Comment:
Concider using the SortOrder type?
dhis-2/dhis-api/src/main/java/org/hisp/dhis/analytics/SortOrder.java

Reply:
Yes!